lesson21/app.js

```diff
@@ -1,200 +1,202 @@
-    ASC: "ASC",
-    NORMAL: "NORMAL",
-  };
-
-  descButton.addEventListener("click", () => {
-    const tbody = document.getElementById("tbody");
-    tbody.remove();
-    if (descButton.getAttribute("data-status") === sortStatus["DESC"]) {
-      descButton.disabled = true;
-      ascButton.disabled = false;
-      descButton.dataset.status = "NORMAL";
-      ascButton.dataset.status = "NORMAL";
-      sortDesc(data);
-    } else {
-      descButton.disabled = false;
-      ascButton.disabled = false;
-      sortInit(data);
-      ascButton.dataset.status = "ASC";
-      descButton.dataset.status = "DESC";
-    }
-  });
+const table = {
+  items: {
+    memberId: "ID",
+    name: "名前",
+    age: "年齢",
+    sex: "性別",
+  },
+  Sort: {
+    Both: "both",
+    Asc: "asc",
+    Desc: "desc",
+  },
+  initData: null,
+};
 
-  ascButton.addEventListener("click", () => {
-    const tbody = document.getElementById("tbody");
-    tbody.remove();
-    if (ascButton.getAttribute("data-status") === sortStatus["ASC"]) {
-      ascButton.disabled = true;
-      descButton.disabled = false;
-      ascButton.dataset.status = "NORMAL";
-      descButton.dataset.status = "NORMAL";
-      sortAsc(data);
-    } else {
-      descButton.disabled = false;
-      ascButton.disabled = false;
-      sortInit(data);
-      ascButton.dataset.status = "ASC";
-      descButton.dataset.status = "DESC";
-    }
-  });
-}
-
-async function initialSettingSort() {
-  const tabletHeader = document.getElementsByTagName("th");
-  const tabletHeaderArray = Array.from(tabletHeader);
-  tabletHeaderArray.forEach((value) => {
-    if (value.hasAttribute("data-sort") == true) {
-      createSortButtons();
-    }
-  });
-}
-
-async function createSortButtons() {
-  let sortsDataAttributes = document.querySelectorAll("[data-sort]");
-  sortsDataAttributes = Array.from(sortsDataAttributes);
-
-  sortsDataAttributes.forEach((value) => {
-    const sortButtonContainerDiv = document.createElement("div");
-    sortButtonContainerDiv.classList.add("sort-button_container");
-
-    const ascButton = document.createElement("button");
-    ascButton.id = "asc";
-    ascButton.classList.add("sort-button", "_asc");
-    ascButton.textContent = "▲";
-    ascButton.dataset.status = "ASC";
-
-    const descButton = document.createElement("button");
-    descButton.id = "desc";
-    descButton.classList.add("sort-button", "_desc");
-    descButton.textContent = "▼";
-    descButton.dataset.status = "DESC";
-
-    sortButtonContainerDiv.appendChild(ascButton);
-    sortButtonContainerDiv.appendChild(descButton);
-    value.appendChild(sortButtonContainerDiv);
-  });
-}
+const sortButton = {
+  memberId: table.Sort.Both,
+};
 
-function sortAsc(data) {
-  const sortResult = [...data].sort(function (a, b) {
-    return a.id - b.id;
-  });
-  createTableBody(sortResult);
-}
+const sortDescFunctor = (key) => (a, b) => {
+  if (a[key] > b[key]) return -1;
+  if (a[key] < b[key]) return 1;
+  return 0;
+};
```

Comment:
Q.
I thought this would be fine.
Or does this function lead to assignment 22?
I would be happy to know.

```suggestion
const sortDescFunctor = (key) => (a, b) => {
  return b[key] - a[key];
};
```

OR

```
const sortDesc = (target, key) => {
  const sortTarget = [...target];
  sortTarget.sort((a, b) => b[key] - a[key]);
  return sortTarget;
};

```

Reply:
Thank you for your question!

> Or does this function lead to assignment 22?

No, it means something much different.

If the value to be sorted is definitely a number
This is still no problem. It's short and easy to read.
```js
const sortDescFunctor = (key) => (a, b) => {
  return b[key] - a[key];
};
```
But...
The value to be sorted this time cannot be said with certainty to be a number. It is possible that strings, NaN, etc. may be included.
So I am writing this way.

[Array.prototype.sort()](https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Global_Objects/Array/sort)


The reason I am splitting the function into two is because I took moriken-sensei's advice the last time I tried it.
For the above reasons, the process in the sort function field became longer, so it was separated for easier reading.

Reply:
I see you considered sorting by something other than numbers!
I hadn't thought of that...That's great!
Thank you so much for taking the time to share this with us!
I have written it down :)!!!